### PR TITLE
Redis integration

### PR DESCRIPTION
**Pull request by mykolaKhlopyk**

### CODE REVIEW

File: rest/src/main/kotlin/systems/ajax/englishstudytelegrambot/repository/WordCashableRepository.kt [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import org.springframework.data.redis.core.ReactiveRedisTemplate
+import org.springframework.stereotype.Repository
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.switchIfEmpty
+import reactor.kotlin.core.publisher.toMono
+import systems.ajax.englishstudytelegrambot.entity.Word
+
+@Suppress("TooManyFunctions")
+interface WordCashableRepository {
```
Comment (ajax-shulepa-a) [inline rest/src/main/kotlin/systems/ajax/englishstudytelegrambot/repository/WordCashableRepository.kt]: why do you need a separate cashable interface? can you reuse the existing interface?

**Review by ajax-shulepa-a — CHANGES_REQUESTED** [review]

File: rest/src/main/kotlin/systems/ajax/englishstudytelegrambot/repository/WordCashableRepository.kt (hunk 1 above)
Comment (mykolaKhlopyk) [reply rest/src/main/kotlin/systems/ajax/englishstudytelegrambot/repository/WordCashableRepository.kt]: yeap, i thought about but wouldn't it to require use a lot of Qualifier?

Comment (ajax-shulepa-a) [reply rest/src/main/kotlin/systems/ajax/englishstudytelegrambot/repository/WordCashableRepository.kt]: I guess managing qualifiers would be easier than managing two interfaces

Comment (mykolaKhlopyk) [reply rest/src/main/kotlin/systems/ajax/englishstudytelegrambot/repository/WordCashableRepository.kt]: Done

File: rest/src/main/kotlin/systems/ajax/englishstudytelegrambot/repository/impl/WordCashableRepositoryImpl.kt [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            }
+
+    override fun isWordBelongsToLibrary(wordSpelling: String, libraryId: ObjectId): Mono<Boolean> =
+        wordRepository.isWordBelongsToLibrary(wordSpelling, libraryId)
+
+    override fun getWord(wordId: ObjectId): Mono<Word> =
+        redisTemplate.opsForValue().get(wordId.createKeyFindWordById())
+            .doOnNext {
+                println(it)
+            }
```
Comment (ajax-shulepa-a) [inline rest/src/main/kotlin/systems/ajax/englishstudytelegrambot/repository/impl/WordCashableRepositoryImpl.kt]: don't commit printlns

File: rest/src/main/kotlin/systems/ajax/englishstudytelegrambot/repository/impl/WordCashableRepositoryImpl.kt [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        getWord(wordId)
+            .map { it.libraryId == libraryId }
+            .flatMap {
+                if (it) {
+                    it.toMono()
+                } else {
+                    Mono.empty()
+                }
+            }
+            .switchIfEmpty {
```
Comment (ajax-shulepa-a) [inline rest/src/main/kotlin/systems/ajax/englishstudytelegrambot/repository/impl/WordCashableRepositoryImpl.kt]:
```suggestion
            .fitler { it }
            .switchIfEmpty {
```

it'll look better but I'm confused with overall logic in this method

**Review by ajax-shulepa-a — APPROVED** [review]

File: rest/src/main/kotlin/systems/ajax/englishstudytelegrambot/repository/impl/WordCashableRepositoryImpl.kt (hunk 3 above)
Comment (mykolaKhlopyk) [reply rest/src/main/kotlin/systems/ajax/englishstudytelegrambot/repository/impl/WordCashableRepositoryImpl.kt]: Done

**Pull request merged**